pandas/core/indexes/base.py

```diff
@@ -539,7 +538,6 @@ def __new__(
 
         klass = cls._dtype_to_subclass(arr.dtype)
 
-        # _ensure_array _may_ be unnecessary once NumericIndex etc are gone
         arr = klass._ensure_array(arr, arr.dtype, copy=False)
```

Comment:
It's not unnecessary?

Reply:
I had to remove “NumericIndex” and had to decide what to do with comment and opted to delete it. IMO _ensure_array could be refactored, but so could a lot of other things in pandas and _ensure_array is not *that* bad IMO anymore compared to other things, so doesn’t merit a comment In the code base IMO.

This was my reasoning, others could of course have made a different judgement call…So it’s not a strong opinion, and I’m fine with it being refactored, but also it not being refactored :-)

Reply:
Yeah no worries, just wasn't sure if this was still being True or not